code/game/objects/machinery/overwatch.dm

```diff
@@ -1216,6 +1246,134 @@ GLOBAL_LIST_EMPTY(active_cas_targets)
 	dat += "<br><A href='?src=\ref[src];operation=back'>{Back}</a>"
 	return dat
 
+/obj/machinery/computer/camera_advanced/overwatch/InterceptClickOn(mob/user, params, atom/object)
+	var/list/pa = params2list(params)
+	if (pa.Find("shift"))
+		var/turf/TU = get_turf(object)
+		if(!send_orders(TU) && current_order)
+			to_chat(usr, "<span class='warning'>Your last order was too recent</span>")
+				
+
+/obj/machinery/computer/camera_advanced/overwatch/proc/send_orders(turf/target_turf)
+	if (world.time > last_order_time + ORDER_COOLDOWN)
+		switch (current_order)
+			if (ATTACK_ORDER)
+				send_attack_orders(target_turf)
+			if (DEFEND_ORDER)
+				send_defend_orders(target_turf)
+			if (RETREAT_ORDER)
+				send_retreat_orders(target_turf)
+			else
+				return TRUE
+		last_order_time = world.time
+		return TRUE
+	else	
+		return FALSE
```

Comment:
early return

Reply:
Early returns in switch? Doesn't that prevent the rest of the code to be unreachable?